pyresample/boundary/base_boundary.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2014-2023 Pyresample developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Define the BaseBoundary class."""
+
+import logging
+
+import numpy as np
+
+from pyresample.boundary.sides import BoundarySides
+
+logger = logging.getLogger(__name__)
+
+
+class BaseBoundary:
+    """Base class for boundary objects."""
+    __slots__ = ["_sides_x", "_sides_y"]
```

Comment:
Can you motivate the use of `__slots__` here? I don't see any real benefit in this class.